docs/translating.md

```diff
@@ -0,0 +1,169 @@
+# UI translation
+
+The website UI is automatically translated in [available languages](TODO: link
+to translation.ts). This is achieved with
+[`react-i18next`](https://react.i18next.com/).
+
+---
+
+<!-- vim-markdown-toc GitLab -->
+
+* [Workflows](#workflows)
+    * [Editing the source code](#editing-the-source-code)
+    * [Adding a new language](#adding-a-new-language)
+* [Architecture](#architecture)
+* [Available scripts](#available-scripts)
+    * [`generate-ui.js`](#generate-uijs)
+    * [`translate-ui.js`](#translate-uijs)
+    * [`translate-md.js`](#translate-mdjs)
+    * [`translate-faq.js`](#translate-faqjs)
+    * [`translate-releases.js`](#translate-releasesjs)
+
+<!-- vim-markdown-toc -->
+
+---
+
+## Workflows
+
+### Editing the source code
+
+When adding a new string containing a content which need to be translated,
+you need to wrap it inside a [`Trans`](https://react.i18next.com/latest/trans-component) React component
+or inside a function call to [`t`](https://react.i18next.com/latest/usetranslation-hook).
+
+Once the content modified,
+
+1. you can update the resource files with the [`generate-ui.js`](#generate-ui.js) script.
+    <details>
+    <summary>Script output example</summary>
+
+    ```
+    > npm run generate:ui
+
+    > nosgestesclimat-site@2.1.0 generate:ui
+    > node scripts/i18n/generate-ui.js
+
+    Static analysis of the source code...
+    Adding missing entries...
+    + Added 2 translations:
+        Cacher l'objectif
+        catégorie complétée
+    ~ Updated 1 translations:
+        components.stats.StatsContent.enSavoirPlus
+    - Missing 2 translations:
+        components.conversation.select.NumberedMosaic.choixAFaire
+        components.conversation.select.NumberedMosaic.choixEnTrop
+    Writting resources in /home/emile/Projects/datagir/nosgestesclimat-site/source/locales/ui/ui-fr.json...
+
+    ```
+    </details>
+
+    For _added_ and _updated_ entries, their translation are automatically
+    retrieved from the source code. However, for missing entries, you need
+    to replace the `'NO_TRANSLATION'` string by the adequate one.
+
+2. then, you need to translate the generated resource file into the other supported languages.
+This can be done automatically by using [`translate-ui.js`](#translate-ui.js).
+
+For non _hardcoded_ content such as the FAQ or Markdown pages.
+You need to modify the corresponding file inside `.source/locales/pages/` (resp. `.source/locales/faq`)
+and then running the script [`translate-md.js`](#translate-md.js) (resp. [`translate-faq.js`](#translate-faq.js))
+to translate it in the remaining languages.
+
+> **Important**:
+>
+> - you need to manually retranslate into french the internal page url.
```

Comment:
>  - you need to manually retranslate into french the internal page url.

I don't understand what this means :sweat_smile:

Reply:
The doc isn't up to date. Indeed, in the first version, I sent Markdown content directly to DeepL but internal links like `[ici](/groupe)` was translated into `[here](/group)`
To solve this, the scripts now used `pandoc` to format Markdown content into HTML, then translating the HTML with DeepL (ignoring `<a>` tags) and then reformatting into Markdown :sweat_smile:

Reply:
Wow ! Be careful to document it. Wasn't simpler to use a regexp to isolate links ? Looks like a huge dependency, you could maybe raise an issue to demand a future fix.

Reply:
> Be careful to document it. Wasn't simpler to use a regexp to isolate links

Yes I'm not fully statisfied with the result of pandoc both in term of perf and translation quality. I'll investigate an other way to do this..